### PR TITLE
mdns: enable CONFIG_ESP_WIFI_ENABLED if CONFIG_SOC_WIFI_SUPPORTED is also enabled (for ESP-IDF <= 5.1)

### PR DESCRIPTION
**Pull request by gabsuren**

### CODE REVIEW

File: components/mdns/mdns.c [hunk 1]
```diff
@@ -26,8 +26,7 @@
 #endif
 
 #if ESP_IDF_VERSION <= ESP_IDF_VERSION_VAL(5, 1, 0)
-// IDF <= v5.1 does not support enabling/disabling esp-wifi
-#define MDNS_ESP_WIFI_ENABLED 1
+#define MDNS_ESP_WIFI_ENABLED CONFIG_SOC_WIFI_SUPPORTED
```
Comment (david-cermak) [inline components/mdns/mdns.c:29]: Is this `CONFIG_SOC_WIFI_SUPPORTED` defined in say `v4.3` ?

Comment (gabsuren) [reply components/mdns/mdns.c:29]: `mdns` as a component should be used only if `idf: ">=5.0"`, why we need to check `v4.3` ?

Comment (me-no-dev) [reply components/mdns/mdns.c:29]: ah so true... mdns is part of IDF before 5.0...

Comment (david-cermak) [reply components/mdns/mdns.c:29]: I see. no more concerns if we support mdns only from `v5.0`

**Review by david-cermak — APPROVED** [review]

**Pull request merged**